cats/cats-sql/src/main/kotlin/com/netflix/spinnaker/cats/sql/cluster/SqlClusteredAgentScheduler.kt

```diff
@@ -211,15 +211,21 @@ class SqlClusteredAgentScheduler(
     }
 
     log.debug("Next agents to run: {}, max: {}", candidateAgentLocks.keys, availableAgents)
+    val shuffledCandidates = mutableMapOf<String, AgentExecutionAction>()
```

Comment:
I don't think it does what you want as it's relying on the map iteration sequence for the randomness. You could just move the `shuffled()` below instead of having that intermediate map.

Reply:
Tests before and after the change showed that agents were randomly executed with the shuffled map, and they were executed always in the same order before. I agree that the intermediate map is not needed and I can use `shuffled` in the original map though 👍